### PR TITLE
feat: Add MediaPlayerElement styles and sample pages

### LINKED ISSUE #1032: [Chefs, TubePlayer] Need a Material style for the MediaPlayer
## What would you like to be added:
In Material the media player should display without using Windows specific elements such as: Acrylic background, Fluent font icons, Win UI scrubber.

## Why is this needed:
Porting compoenent to Material

## For which Platform:

- [x] UWP
- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] macOS

## Anything else we need to know?
Here is the link to the Figma File aka Source of truth :)
https://www.figma.com/file/trM3i5qUPmFBcnxWXXmnuF/branch/LXYP010W9oJANcAFpbDbef/Uno-Platform-Material-Toolkit?type=design&node-id=3082%3A107842&t=fdxl4yDystednvDw-1

### PR DESCRIPTION
**Pull request by nguyeenvy**
closes #1032 

﻿GitHub Issue: #1032 

## PR Type

What kind of change does this PR introduce?
- Feature

## Description
Add MediaPlayerElement styles and sample pages

## PR Checklist 
- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [X] Tested UWP
- [X] Tested iOS
- [X] Tested Android
- [X] Tested WASM
- [ ] Tested MacOS
- [X] Contains **No** breaking changes

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/Slider.xaml
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/Slider.xaml:1]: Are these changes still required?

File: .github/workflows/azure-static-webapp.yml
Comment (kazo0) [inline .github/workflows/azure-static-webapp.yml:1]: Let's rebase this off of later master and we shouldn't need some of these changes

Comment (nguyeenvy) [reply .github/workflows/azure-static-webapp.yml:1]: I just rebased this branch off of latest master. The changes in the Slider.xaml have been removed.

File: src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml [hunk 1]
```diff
@@ -0,0 +1,653 @@
+﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
```
Comment (agneszitte) [inline src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml:2]: Format XAML using XAML styler + tabs

Comment (Marc-Antoine-Soucy) [reply src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml:2]: I think we don't have access to that comment. Unless it's just me. But I will help her set up the xaml style if its not done.
image

File: src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml [hunk 2]
```diff
@@ -0,0 +1,653 @@
...
+														  Style="{StaticResource MediaTransportControlsAppBarButtonStyle}"
+														  MediaTransportControlsHelper.DropoutOrder="15"
+														  Visibility="Collapsed">
+												<AppBarButton.Icon>
+													<FontIcon Glyph="&#xED1E;" />
+												</AppBarButton.Icon>
+											</AppBarButton>
+											<AppBarButton x:Name="AudioTracksSelectionButton"
+														  Style="{StaticResource MediaTransportControlsAppBarButtonStyle}"
+														  MediaTransportControlsHelper.DropoutOrder="13"
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml:521]: For now let's just add `not_win:` prefixes to all of the `MediaTransportControlHelper.DropoutOrder` properties. It requires at least WinAppSDK v1.2 and we are still on 1.1.0 for Material and 1.1.5 for Toolkit

Add a comment in the file about why the not_win is there for now

File: src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml [hunk 3]
```diff
@@ -0,0 +1,655 @@
+﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+					xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+					xmlns:win="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+					xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
+					xmlns:todo="what should be done"
+					xmlns:contract7Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,7)"
+					xmlns:contract7NotPresent="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractNotPresent(Windows.Foundation.UniversalApiContract,7)"
+					xmlns:contract8Present="http://schemas.microsoft.com/winfx/2006/xaml/presentation?IsApiContractPresent(Windows.Foundation.UniversalApiContract,8)"
+					mc:Ignorable="todo">
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml:9]:
```suggestion
					xmlns:not_win="http://uno.ui/not_win"
					mc:Ignorable="todo not_win">
```

File: src/samples/UWP/Uno.Themes.Samples.Shared/Content/NestedSamples/MediaPlayerElementSample_NestedPage5.xaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+							{
+							var mediaPlaybackList = new MediaPlaybackList();
+							
+							mediaPlaybackList.Items.Add(new MediaPlaybackItem(MediaSource.CreateFromUri(new Uri("https://uno-assets.platform.uno/tests/videos/Mobile_Development_in_VS_Code_with_Uno_Platform_orDotNetMAUI.mp4"))));
+							mediaPlaybackList.Items.Add(new MediaPlaybackItem(MediaSource.CreateFromUri(new Uri("https://uno-assets.platform.uno/tests/audio/Getting_Started_with_Uno_Platform_and_Visual_Studio_Code.mp3"))));
+							mediaPlaybackList.Items.Add(new MediaPlaybackItem(MediaSource.CreateFromUri(new Uri("https://uno-assets.platform.uno/tests/videos/Getting_Started_with_Uno_Platform_and_Visual_Studio_Code.mp4"))));
+							
+							MediaPlayerElementSample5.MediaPlayer.Source = mediaPlaybackList;
+							}
+						-->
```
Comment (pictos) [inline src/samples/UWP/Uno.Themes.Samples.Shared/Content/NestedSamples/MediaPlayerElementSample_NestedPage5.xaml:89]: is this needed?

**Review by pictos — COMMENTED** [review]
Looks good, I've just a small comment

**Review by Xiaoy312 — APPROVED** [review]

File: src/samples/UWP/Uno.Themes.Samples.Shared/Content/NestedSamples/MediaPlayerElementSample_NestedPage5.xaml (hunk 4 above)
Comment (agneszitte) [reply src/samples/UWP/Uno.Themes.Samples.Shared/Content/NestedSamples/MediaPlayerElementSample_NestedPage5.xaml:89]: yeah, you can remove this @nguyeenvy as we are adding it only for Uno.Gallery to showcase the code-behind in XamlDisplay as it is a showcase app. But Uno.Themes sample for tests does not need it.

**Review by kazo0 — COMMENTED** [review]
It looks like we're going to need to bump the WinAppSDK version to at least 1.2 in order to have support for MediaTransportControls (https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.mediatransportcontrols?view=windows-app-sdk-1.3#applies-to)

FYI @agneszitte / @Xiaoy312

**Comment by agneszitte** [comment]
Also when this PR is merged @kazo0, @Xiaoy312, @nguyeenvy we will need to add more samples for the Gallery app

File: src/samples/UWP/Uno.Themes.Samples.Shared/Content/NestedSamples/MediaPlayerElementSample_NestedPage2.xaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+								AreTransportControlsEnabled="True"
+								AutoPlay="False"
+								MaxWidth="720">
+
+				<MediaPlayerElement.TransportControls>
+					<MediaTransportControls IsCompact="True" />
+				</MediaPlayerElement.TransportControls>
+			</MediaPlayerElement>
+
+			<!-- Source="ms-appx:///Assets/Videos/Getting_Started_with_Uno_Platform_for_Figma.mp4" -->
```
Comment (pictos) [inline src/samples/UWP/Uno.Themes.Samples.Shared/Content/NestedSamples/MediaPlayerElementSample_NestedPage2.xaml:56]: can we remove this comment?

**Comment by kazo0** [comment]
@nguyeenvy let's bump the versions of `Microsoft.WindowsAppSDK` to 1.2.0

File: src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml [hunk 6]
```diff
@@ -0,0 +1,673 @@
...
+	<x:String x:Key="VolumePathData">m4,9.17c-.55,0-1,.42-1,.94v3.78c0,.52.45.94,1,.94h2.59c.27,0,.52.1.71.28l3,2.83c.63.6,1.71.17,1.71-.67V6.72c0-.84-1.08-1.26-1.71-.67l-3,2.83c-.19.18-.44.28-.71.28h-2.59Zm12.5,2.83c0-1.12-.46-2.14-1.21-2.9-.51-.51-1.29-.08-1.29.62v4.56c0,.7.78,1.13,1.29.62.75-.76,1.21-1.77,1.21-2.9Zm-1.3-7.95c-.6-.22-1.2.24-1.2.84v.06c0,.42.29.78.7.94,2.52,1,4.3,3.36,4.3,6.1s-1.77,5.1-4.3,6.1c-.41.16-.7.52-.7.94v.06c0,.61.6,1.06,1.2.84,3.39-1.22,5.8-4.32,5.8-7.95s-2.41-6.73-5.8-7.95Z</x:String>
+
+	<Style TargetType="MediaTransportControls"
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/MediaPlayerElement.xaml:94]: We should follow the pattern that we are using for the rest of the controls when defining the default styles. Like [here in PasswordBox](https://github.com/unoplatform/Uno.Themes/blob/ab039e406f5725a88e6502fb80b773f5e3d11d91/src/library/Uno.Material/Styles/Controls/v2/PasswordBox.xaml#L579-L581).

The DefaultMediaTransportControlsStyle should be defined at the bottom of the style XAML file.

And then we should add the implicit style based on DefaultMediaTransportControlsStyle in the [_Resources.xaml](https://github.com/unoplatform/Uno.Themes/blob/ab039e406f5725a88e6502fb80b773f5e3d11d91/src/library/Uno.Material/Styles/Controls/v2/_Resources.xaml#L31)

**Comment by nguyeenvy** [comment]
@kazo0 Microsoft.WindowsAppSDK has been successfully bumped to 1.2.0

**Comment by agneszitte** [comment]
> Also when this PR is merged @kazo0, @Xiaoy312, @nguyeenvy we will need to add more samples for the Gallery app

I have opened an issue for it https://github.com/unoplatform/Uno.Gallery/issues/909

**Review by kazo0 — APPROVED** [review]

**Comment by sasakrsmanovic** [comment]
@nguyeenvy Thanks for that PR. I don't recall if I or @mtmattei had sent you an Uno Platform t-shirt as a small token of appreciation for the contribution. If not , please get in touch with me at sasha at platform . uno. Thanks again!
